tests/unit/src/main/scala/tests/BaseScalaCliSuite.scala

```diff
@@ -0,0 +1,304 @@
+      else if (
+        params.getType == MessageType.ERROR && params.getMessage.startsWith(
+          "Error importing Scala CLI project "
+        )
+      )
+        importedPromise.failure(
+          new Exception(s"Error importing project: $params")
+        )
+      else
+        previousShowMessageHandler(params)
+    }
+    val previousShowMessageRequestHandler =
+      server.client.showMessageRequestHandler
+    server.client.showMessageRequestHandler = { params =>
+      def useBsp = Files.exists(
+        server.server.workspace.resolve(".bsp/scala-cli.json").toNIO
+      )
+      if (params == Messages.ImportScalaScript.params())
+        Some(
+          new MessageActionItem(
+            if (useBsp)
+              Messages.ImportScalaScript.dismiss
+            else
+              Messages.ImportScalaScript.doImportScalaCli
+          )
+        )
+      else if (params == Messages.ImportAllScripts.params())
+        Some(
+          new MessageActionItem(
+            if (useBsp)
+              Messages.ImportAllScripts.dismiss
+            else
+              Messages.ImportAllScripts.importAll
+          )
+        )
+      else
+        previousShowMessageRequestHandler(params)
+    }
+  }
+
+  private def manualLayout =
+    s"""/metals.json
+       |{
+       |  "a": {
+       |    "scalaVersion": "$scalaVersion"
+       |  }
+       |}
+       |
+       |""".stripMargin
+
+  private def escape(s: String): String =
+    s.replace("\\", "\\\\")
+  private def bspLayout =
+    s"""/.bsp/scala-cli.json
+       |{
+       |  "name": "scala-cli",
+       |  "argv": [
+       |    "${escape(ScalaCli.javaCommand)}",
+       |    "-cp",
+       |    "${escape(ScalaCli.scalaCliClassPath().mkString(File.pathSeparator))}",
+       |    "${ScalaCli.scalaCliMainClass}",
+       |    "bsp",
+       |    "."
+       |  ],
+       |  "version": "${BuildInfo.scalaCliVersion}",
+       |  "bspVersion": "2.0.0",
+       |  "languages": [
+       |    "scala",
+       |    "java"
+       |  ]
+       |}
+       |
+       |/.scala-build/ide-inputs.json
+       |{
+       |  "args": [
+       |    "."
+       |  ]
+       |}
+       |
+       |""".stripMargin
+
+  private def scalaCliInitialize(
+      useBsp: Boolean
+  )(layout: String): Future[InitializeResult] = {
+    if (!useBsp)
+      importedPromise = Promise[Unit]()
+    initialize(
+      (if (useBsp) bspLayout else manualLayout) + layout
+    )
+  }
+
+  private def waitForImport(useBsp: Boolean): Future[Unit] =
+    if (useBsp) Future.successful(())
+    else
+      Future
+        .firstCompletedOf(
+          List(
+            importedPromise.future,
+            timeout("import timeout", 180.seconds),
+          )
+        )
+
+  private lazy val isAtLeastJava11: Boolean = {
+    val versionOpt = sys.props.get("java.version")
+    val majorOpt = versionOpt.map(_.split('.').head.toInt)
+    majorOpt.exists(_ >= 11)
+  }
+
+  if (isAtLeastJava11)
```

Comment:
Why at least JDK 11?

Reply:
I can't recall… Running Scala CLI with Java 17, as done in `ScalaCli.scala`, might actually have fixed the errors I was seeing. Going to remove that in the next push, we'll see how it goes.